workspaces/arborist/lib/arborist/build-ideal-tree.js

```diff
@@ -1299,8 +1299,8 @@ This is a one-time fix-up, please be patient...
         continue
       }
 
-      const parentEdge = node.parent.edgesOut.get(edge.name)
-      const { isProjectRoot, isWorkspace } = node.parent.sourceReference
+      const parentEdge = node.parent?.edgesOut.get(edge.name) 
+      const { isProjectRoot, isWorkspace } = node.parent?.sourceReference || {}
```

Comment:
This line is missing coverage now.

Reply:
Our test coverage for this repo is at 100%, you added a branching path with the `||` that isn't covered by tests.

Reply:
Add a test that covers this scenario. This is the root of the bug, we need to isolate exactly what gets the code in this situation so we can test that we've fixed it.